evap/staff/importers/enrollment.py

```diff
@@ -0,0 +1,764 @@
+    student_first_name: str
+    student_email: str
+
+    evaluation_course_type_name: str
+    evaluation_is_graded: str
+    evaluation_name_de: str
+    evaluation_name_en: str
+
+    responsible_title: str
+    responsible_last_name: str
+    responsible_first_name: str
+    responsible_email: str
+
+
+@dataclass
+class EnrollmentParsedRow:
+    """
+    Representation of an Enrollment Row after parsing the data into the resulting data structures.
+    For example, the course degree will already be resolved in here.  This is the data structure we want to work with.
+    """
+
+    location: ExcelFileLocation
+
+    student_data: UserData
+    responsible_data: UserData
+    course_data: CourseData
+
+
+class EnrollmentInputRowMapper:
+    def __init__(self, importer_log: ImporterLog):
+        self.importer_log: ImporterLog = importer_log
+
+        self.course_type_mapper: CourseTypeImportMapper = CourseTypeImportMapper()
+        self.degree_mapper: DegreeImportMapper = DegreeImportMapper()
+        self.is_graded_mapper: IsGradedImportMapper = IsGradedImportMapper()
+
+        self.invalid_degrees_tracker: Optional[FirstLocationAndCountTracker] = None
+        self.invalid_course_types_tracker: Optional[FirstLocationAndCountTracker] = None
+        self.invalid_is_graded_tracker: Optional[FirstLocationAndCountTracker] = None
+
+    def map(self, rows: Iterable[EnrollmentInputRow]) -> Iterable[EnrollmentParsedRow]:
+        self.invalid_degrees_tracker = FirstLocationAndCountTracker()
+        self.invalid_course_types_tracker = FirstLocationAndCountTracker()
+        self.invalid_is_graded_tracker = FirstLocationAndCountTracker()
+
+        result_rows = [self._map_row(row) for row in rows]
+        self._log_aggregated_messages()
+        return result_rows
+
+    def _map_row(self, row: EnrollmentInputRow) -> EnrollmentParsedRow:
+        assert self.invalid_degrees_tracker is not None
+        assert self.invalid_course_types_tracker is not None
+        assert self.invalid_is_graded_tracker is not None
+
+        student_data = UserData(
+            first_name=row.student_first_name,
+            last_name=row.student_last_name,
+            email=row.student_email,
+            title="",
+        )
+
+        responsible_data = UserData(
+            first_name=row.responsible_first_name,
+            last_name=row.responsible_last_name,
+            title=row.responsible_title,
+            email=row.responsible_email,
+        )
+
+        try:
+            degrees = self.degree_mapper.degree_set_from_import_string(row.evaluation_degree_names)
+        except DegreeImportMapper.InvalidDegreeNamesException as e:
+            degrees = invalid_value
+            for invalid_degree in e.invalid_degree_names:
+                self.invalid_degrees_tracker.add_location_for_key(row.location, invalid_degree)
+
+        try:
+            course_type: Union[InvalidValue, CourseType] = self.course_type_mapper.course_type_from_import_string(
```

Comment:
This is only needed because of the assignment below, how do you feel about declaring the three variables that can be invalid just before the first `try` (this is also shorter):

```python
degrees: ...
course_type: Union[CourseType, InvalidValue]
is_graded: Union[bool, InvalidValue]
```